src/plugins/blocksmith/components/blocksmith/index.jsx

```diff
@@ -1388,29 +1388,30 @@ class Blocksmith extends React.Component {
         ...sidebarHolderStyle,
         transform: `scale(${isDocumentClick ? 0 : 1})`,
       },
-    }), this.createPopoverToggleEvent);
-  }
-
-  createPopoverToggleEvent() {
-    const { isSidebarOpen } = this.state;
+    }), () => {
+      const { isSidebarOpen } = this.state;
 
-    if (isSidebarOpen) {
-      document.addEventListener('click', this.handleTogglePopover);
-    } else {
-      document.removeEventListener('click', this.handleTogglePopover);
-    }
+      if (isSidebarOpen) {
+        document.addEventListener('click', this.handlePopoverClick);
+      } else {
+        document.removeEventListener('click', this.handlePopoverClick);
+      }
+    });
   }
 
-  handleTogglePopover(e) {
-    const { isSidebarOpen } = this.state;
-
+  /**
+   * This is needed to check if a click event occured anywhere outside of the popover.
+   * When a click event occurs outside of the popover, the popover will be toggled to close.
+   *
+   * @param {SyntheticKeyboardEvent} e - a synthetic keyboard event
```

Comment:
@cohencodes i dont think this is synthetic. you get a synthetic when its a react click handler